pkg/lightning/backend/local/local.go

```diff
@@ -2603,8 +2709,22 @@ func (w *Writer) appendRowsSorted(kvs []common.KvPair) error {
 		w.writer = writer
 		w.writer.minKey = append([]byte{}, kvs[0].Key...)
 	}
-	for _, pair := range kvs {
-		w.batchSize += int64(len(pair.Key) + len(pair.Val))
+	if w.duplicateDetection {
```

Comment:
Is it possible to move this encode to `tableKvEncoder` in `TakeKvPairs` so we need do it in multi functions. 
BTW, can we make it a interface like `KVAdapter` that provide `encode` and `decode` so maybe in the future, we can easily implement other kind of Adapters? Maybe we can also avoiding check the `duplicateDetection` flag in multi place

Reply:
+1

Reply:
We do kv checksum after `tableKvEncoder`,  so we may not encode key in `tableKvEncoder `.

`KVAdapter` sounds very reasonable and elegant. I will implement it. Thanks.

Reply:
I add a `KeyAdapter` in https://github.com/pingcap/br/pull/1126/commits/9140be0fa594262d3fb6f279fe5b655a75bf9b9d. Is this ok? @glorv